src/accelerate/accelerator.py

```diff
@@ -2854,6 +2854,10 @@ def save_model(
                 raise RuntimeError("You can't save the model since some parameters are on the meta device.")
             state_dict = self.get_state_dict(model)
 
+        # Case: DeepSpeed zero3 gets gathered and `state_dict` is empty
+        if state_dict is None:
+            return
```

Comment:
We can maybe make it more specific to deepspeed ? 
In their code, this is how they do this: https://github.com/microsoft/DeepSpeed/blob/f3943cf9109226ed3ecf2d5dbb639a11cd925555/deepspeed/runtime/engine.py#L3414
Seems like the model only gets saved on rank 0

Reply:
I went more generic just in case somehow this happens in other places that we don't know yet (so we enable and don't block them)